packages/core/src/state-management/state-manager.ts

```diff
@@ -217,20 +218,14 @@ export class StateManager {
    * @param model - Model Stream ID
    */
   async handlePubsubUpdate(streamId: StreamID, tip: CID, model?: StreamID): Promise<void> {
-    let state$ = await this.fromMemoryOrStore(streamId)
+    const state$ = await this.fromMemoryOrStore(streamId)
     const shouldIndex = model && this._index.shouldIndexStream(model)
     if (!shouldIndex && !state$) {
       // stream isn't pinned or indexed, nothing to do
       return
     }
 
-    if (!state$) {
-      state$ = await this.load(streamId)
-    }
-    this.executionQ.forStream(streamId).add(async () => {
-      await this._handleTip(state$, tip)
-    })
-    await this.indexStreamIfNeeded(state$)
+    await this.applyCommit(streamId, tip, {})
```

Comment:
tests are currently failing consistently (`ceramic-anchor.test.ts - test the same doc anchored twice (different Ceramic instances), first one wins)`).  Reverting the changes to this one function makes it pass again consistently.  

The only difference between what this function was doing before and what applyCommit does is that when the state is already in the cache or state store, we get it without going through the loading queue, whereas when we use applyCommit we always go through the loading queue.  That should not affect correctness however.  The fact that that small change is causing the test to break suggests to me that there is a bug in the queue logic somewhere.  I wonder if this is related to the queue filling up errors we've been seeing.

Reply:
lol I'm dumb.  stateManager.applyCommit takes a commit body, not a CID, but I was passing it the commit CID.  Has nothing to do with the queues